tianshou/utils/net/common.py

```diff
@@ -209,58 +233,71 @@ def __init__(
         self.device = device
         self.softmax = softmax
         self.num_atoms = num_atoms
+        self.Q: Optional[MLP] = None
+        self.V: Optional[MLP] = None
+
         input_dim = int(np.prod(state_shape))
         action_dim = int(np.prod(action_shape)) * num_atoms
         if concat:
             input_dim += action_dim
         self.use_dueling = dueling_param is not None
         output_dim = action_dim if not self.use_dueling and not concat else 0
         self.model = MLP(
-            input_dim, output_dim, hidden_sizes, norm_layer, norm_args, activation,
-            act_args, device, linear_layer
+            input_dim,
+            output_dim,
+            hidden_sizes,
+            norm_layer,
+            norm_args,
+            activation,
+            act_args,
+            device,
+            linear_layer,
         )
-        self.output_dim = self.model.output_dim
         if self.use_dueling:  # dueling DQN
-            q_kwargs, v_kwargs = dueling_param  # type: ignore
-            q_output_dim, v_output_dim = 0, 0
-            if not concat:
-                q_output_dim, v_output_dim = action_dim, num_atoms
-            q_kwargs: Dict[str, Any] = {
-                **q_kwargs, "input_dim": self.output_dim,
-                "output_dim": q_output_dim,
-                "device": self.device
-            }
-            v_kwargs: Dict[str, Any] = {
-                **v_kwargs, "input_dim": self.output_dim,
-                "output_dim": v_output_dim,
-                "device": self.device
+            kwargs_update = {
+                "input_dim": self.model.output_dim,
+                "device": self.device,
             }
+            # Important: don't change the original dict (e.g., don't use .update())
+            q_kwargs = {**dueling_param[0], **kwargs_update}  # type: ignore
+            v_kwargs = {**dueling_param[1], **kwargs_update}  # type: ignore
```

Comment:
```suggestion
            q_kwargs = cast(dict[str, Any], {**dueling_param[0], **kwargs_update})
            v_kwargs = cast(dict[str, Any], {**dueling_param[1], **kwargs_update})
```

Reply:
Done

Reply:
An assert was missing there, no need for cast